src/main/java/com/iabtcf/v2/BitVectorTCModelV2.java

```diff
@@ -0,0 +1,348 @@
+        this.vendorListVersion =
+                coreBitVector.readUnsignedInt(VENDOR_LIST_VERSION_OFFSET, SHORT.length());
+        this.policyVersion = coreBitVector.readUnsignedInt(TCF_POLICY_VERSION_OFFSET, TINY_INT.length());
+        this.isServiceSpecific = coreBitVector.readBit(IS_SERVICE_SPECIFIC_OFFSET);
+        this.useNonStandardStacks = coreBitVector.readBit(USE_NON_STANDARD_STACKS_OFFSET);
+        this.specialFeaturesOptInts =
+                fillSet(SPECIAL_FEATURE_OPT_INS_OFFSET, SPECIAL_FEATURE_OPT_INS_LENGTH, coreBitVector);
+        this.purposesConsent = fillSet(PURPOSES_CONSENT_OFFSET, PURPOSES_CONSENT_LENGTH, coreBitVector);
+        this.purposesLITransparency =
+                fillSet(PURPOSE_LI_TRANSPARENCY_OFFSET, PURPOSE_LI_TRANSPARENCY_LENGTH, coreBitVector);
+        this.isPurposeOneTreatment = coreBitVector.readBit(PURPOSE_ONE_TREATMENT_OFFSET);
+        this.publisherCountryCode = coreBitVector.readStr(PUBLISHER_CC_OFFSET, CHAR.length() * 2);
+
+        int currentPointer = PUBLISHER_CC_OFFSET + (CHAR.length() * 2); // publisher cc offset
+        this.vendorConsents = new TreeSet<>();
+        currentPointer = this.fetchSet(this.vendorConsents, currentPointer, coreBitVector);
+
+        this.vendorLegitimateInterests = new TreeSet<>();
+        currentPointer = this.fetchSet(this.vendorLegitimateInterests, currentPointer, coreBitVector);
+
+        this.publisherRestrictions = new ArrayList<>();
+        this.fillPublisherRestrictions(publisherRestrictions, currentPointer, coreBitVector);
+
+        this.disclosedVendors = new TreeSet<>();
+        this.allowedVendors = new TreeSet<>();
+        this.publisherPurposesConsent = new TreeSet<>();
+        this.publisherPurposesLITransparency = new TreeSet<>();
+        this.customPurposesConsent = new TreeSet<>();
+        this.customPurposesLITransparency = new TreeSet<>();
+    }
+
+    private BitVectorTCModelV2(BitVector coreVector, BitVector secondPart) {
+        this(coreVector);
+        fillRemainingVector(secondPart);
+    }
+
+    private BitVectorTCModelV2(BitVector coreVector, BitVector secondPart, BitVector thirdPart) {
+        this(coreVector, secondPart);
+        fillRemainingVector(thirdPart);
+    }
+
+    public static BitVectorTCModelV2 fromBitVector(BitVector coreBitVector) {
+        return new BitVectorTCModelV2(coreBitVector);
+    }
+
+    public static BitVectorTCModelV2 fromBitVector(
+            BitVector coreBitVector, BitVector remainingVector) {
+        return new BitVectorTCModelV2(coreBitVector, remainingVector);
+    }
+
+    public static BitVectorTCModelV2 fromBitVector(
+            BitVector coreBitVector, BitVector secondVector, BitVector thirdVector) {
+        return new BitVectorTCModelV2(coreBitVector, secondVector, thirdVector);
+    }
+
+    private int fillRemainingVector(BitVector bitVector) {
+        int currentPointer = 0;
+        int segmentType = bitVector.readUnsignedInt(SEGMENT_TYPE_OFFSET, 3);
```

Comment:
Just because I had more time to review and I am absolutely being picky here, maybe make the number 3 a constant using variable name like SEGMENT_TYPE_OFFSET_LENGTH ?

Reply:
done